Demos/01-create-app/graph-tutorial/packages.config

```diff
@@ -1,50 +1,35 @@
-  <package id="Microsoft.IdentityModel.Tokens" version="5.2.4" targetFramework="net47" />
-  <package id="Microsoft.jQuery.Unobtrusive.Validation" version="3.2.10" targetFramework="net47" />
-  <package id="Microsoft.Owin" version="4.0.0" targetFramework="net47" />
-  <package id="Microsoft.Owin.Host.SystemWeb" version="4.0.0" targetFramework="net47" />
-  <package id="Microsoft.Owin.Security" version="4.0.0" targetFramework="net47" />
-  <package id="Microsoft.Owin.Security.Cookies" version="4.0.0" targetFramework="net47" />
-  <package id="Microsoft.Owin.Security.OpenIdConnect" version="4.0.0" targetFramework="net47" />
-  <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" targetFramework="net47" />
-  <package id="Modernizr" version="2.8.3" targetFramework="net47" />
-  <package id="Newtonsoft.Json" version="11.0.2" targetFramework="net47" />
-  <package id="Owin" version="1.0" targetFramework="net47" />
-  <package id="popper.js" version="1.14.3" targetFramework="net47" />
-  <package id="System.Diagnostics.DiagnosticSource" version="4.5.0" targetFramework="net47" />
-  <package id="System.IdentityModel.Tokens.Jwt" version="5.2.4" targetFramework="net47" />
-  <package id="System.IO" version="4.3.0" targetFramework="net47" />
-  <package id="System.Net.Http" version="4.3.3" targetFramework="net47" />
-  <package id="System.Runtime" version="4.3.0" targetFramework="net47" />
-  <package id="System.Security.Cryptography.Algorithms" version="4.3.1" targetFramework="net47" />
-  <package id="System.Security.Cryptography.Encoding" version="4.3.0" targetFramework="net47" />
-  <package id="System.Security.Cryptography.Primitives" version="4.3.0" targetFramework="net47" />
-  <package id="System.Security.Cryptography.X509Certificates" version="4.3.2" targetFramework="net47" />
-  <package id="WebGrease" version="1.6.0" targetFramework="net47" />
+  <package id="Antlr" version="3.5.0.2" targetFramework="net472" />
+  <package id="bootstrap" version="4.3.1" targetFramework="net472" />
+  <package id="jQuery" version="3.3.1" targetFramework="net472" />
+  <package id="jQuery.Validation" version="1.17.0" targetFramework="net472" />
+  <package id="Microsoft.AspNet.Mvc" version="5.2.7" targetFramework="net472" />
+  <package id="Microsoft.AspNet.Razor" version="3.2.7" targetFramework="net472" />
+  <package id="Microsoft.AspNet.TelemetryCorrelation" version="1.0.5" targetFramework="net472" />
+  <package id="Microsoft.AspNet.Web.Optimization" version="1.1.3" targetFramework="net472" />
+  <package id="Microsoft.AspNet.WebPages" version="3.2.7" targetFramework="net472" />
+  <package id="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" version="2.0.1" targetFramework="net472" />
+  <package id="Microsoft.Graph" version="1.12.0" targetFramework="net472" />
+  <package id="Microsoft.Graph.Core" version="1.12.0" targetFramework="net472" />
+  <package id="Microsoft.Identity.Client" version="2.7.1" targetFramework="net472" />
```

Comment:
is there any reason you didn't use 3.0.0-preview?

Reply:
Two reasons: (1) from previous discussions with you, you told us to not upgrade MSAL in any of these until you said to do it & (2) as a general rule for this project with OfficeDev, we have not used preview tech in any modules.

Reply:
OK...just to be clear, all of v2 is -preview, it was a mistake that the -preview tag was removed. This is not your fault you do not know this. For now this is great, we will have to update once the -preview tag is removed from v3.0 around Build time frames.

Reply:
OK... we are planning to update all the modules we're refreshing to v3 & to use the new app registration process in the AAD portal once the preview tags are all removed after Build.